File: apps/web/src/lib/types/platform-user.ts
export type PlatformRole = "super_admin" | "operations_admin" | "finance_admin" | "support_admin";
export type PlatformUserStatus = "active" | "disabled";

export interface PlatformUser {
  id: string;
  email: string;
  full_name: string;
  role: PlatformRole;
  status: PlatformUserStatus;
  is_platform: boolean;
  last_login_at: string | null;
  created_at: string;
}

export interface PlatformUserCreateRequest {
  email: string;
  full_name: string;
  role: PlatformRole;
  password: string;
}

export interface PlatformUserUpdateRequest {
  full_name?: string;
  role?: PlatformRole;
  status?: PlatformUserStatus;
}

export interface PlatformUserResetPasswordRequest {
  new_password: string;
}

export interface PlatformUserListResponse {
  items: PlatformUser[];
  total: number;
}

File: apps/api/schemas/platform_user.py
"""Platform user request/response schemas."""

import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict, EmailStr, Field

from apps.api.models.enums import PlatformRole, UserStatus


class PlatformUserResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    email: str
    full_name: str
    role: PlatformRole
    status: UserStatus
    email_verified: bool
    last_login_at: datetime | None = None
    created_at: datetime
    updated_at: datetime


class PlatformUserListResponse(BaseModel):
    items: list[PlatformUserResponse]
    total: int


class PlatformUserCreateRequest(BaseModel):
    email: EmailStr
    full_name: str = Field(..., min_length=1, max_length=255)
    role: PlatformRole
    password: str = Field(..., min_length=8, max_length=128)


class PlatformUserUpdateRequest(BaseModel):
    """PATCH — all fields optional."""

    full_name: str | None = Field(default=None, min_length=1, max_length=255)
    role: PlatformRole | None = None
    status: UserStatus | None = None


class PlatformUserResetPasswordRequest(BaseModel):
    new_password: str = Field(..., min_length=8, max_length=128)
